sdk/storage/Azure.Storage.DataMovement/src/ServiceToServiceTransferJob.cs

```diff
@@ -171,31 +171,49 @@ private async IAsyncEnumerable<JobPartInternal> GetStorageResourcesAsync()
                 }
 
                 StorageResource current = enumerator.Current;
-                if (!existingSources.Contains(current.Uri))
+
+                if (current.IsContainer)
                 {
+                    // Create sub-container
                     string containerUriPath = _sourceResourceContainer.Uri.GetPath();
-                    string sourceName = string.IsNullOrEmpty(containerUriPath)
+                    string subContainerPath = string.IsNullOrEmpty(containerUriPath)
                         ? current.Uri.GetPath()
                         : current.Uri.GetPath().Substring(containerUriPath.Length + 1);
-
-                    ServiceToServiceJobPart part;
-                    try
-                    {
-                        part = await ServiceToServiceJobPart.CreateJobPartAsync(
-                            job: this,
-                            partNumber: partNumber,
-                            sourceResource: (StorageResourceItem)current,
-                            destinationResource: _destinationResourceContainer.GetStorageResourceReference(sourceName))
-                            .ConfigureAwait(false);
-                        AppendJobPart(part);
-                    }
-                    catch (Exception ex)
+                    StorageResourceContainer subContainer =
+                        _destinationResourceContainer.GetChildStorageResourceContainer(subContainerPath);
+                    await subContainer.CreateIfNotExistsAsync().ConfigureAwait(false);
+                }
+                else
+                {
+                    if (!existingSources.Contains(current.Uri))
                     {
-                        await InvokeFailedArgAsync(ex).ConfigureAwait(false);
-                        yield break;
+                        string containerUriPath = _sourceResourceContainer.Uri.GetPath();
+                        string sourceName = string.IsNullOrEmpty(containerUriPath)
+                            ? current.Uri.GetPath()
+                            : current.Uri.GetPath().Substring(containerUriPath.Length + 1);
+
+                        // Because AsyncEnumerable doesn't let us know which storage resource is the last resource
+                        // we only yield return when we know this is not the last storage resource to be listed
+                        // from the container.
```

Comment:
nit: You can remove these comments since we don't track final part anymore. I think I removed them in my last PR but maybe I missed one or they came back in the mrege.

Reply:
done

Reply:
also done for the StreamToUri too